proto/gds/global/v1/global.proto

```diff
@@ -0,0 +1,94 @@
+// that the replica only wants to consider the objects described rather than the entire
+// namespace (e.g. objects that may have been created outside the set of objects
+// described). The partial flag is typically used as a mechanism to fetch specific
+// objects that are known to be later from the remote replica. The namespaces array
+// specifies which namespaces should be considered in gossip allowing for partial
+// replication by namespace if necessary. If the namespaces array is empty, then all
+// namespaces are considered during Gossip.
+message VersionVectors {
+    // Version vectors of objects without data.
+    repeated Object objects = 1;
+
+    // A request to consider only the objects specified in the version vector and not
+    // the entire namespace (e.g. ignore objects that are not specified).
+    bool partial = 2;
+
+    // Limit the comparison only to the specified namespace(s). If not set or empty,
+    // then all namespaces are considered.
+    repeated string namespaces = 3;
+}
+
+// Updates returns data to repair entropy (e.g. to repair the divergence caused by
+// updates). Updates should only contain the objects that have later versions and the
+// objects should have their data fields populated.
+message Updates {
+    repeated Object objects = 1;
+}
+
+// An Object is a generic representation of a replicated piece of data. At the top level
+// it contains enough metadata in order to detect changes during anti-entropy,
+// specifically, given two objects, which is the later object (or does it need to be
+// created or deleted). When used in VersionVectors only the metadata of the object is
+// supplied. When passed via Updates, then the full data of the object is populated.
+//
+// For research purposes, this anti-entropy mechanism tracks the region and owner of
+// each object to determine provinence and global interactions. Because this is side
+// channel information that may not necessarily be stored with the object data, it is
+// recommended that an objects table be kept for fast lookups of object versions.
+// Additionally, it is recommended that a history table be maintained locally so that
+// Object versions can be rolled back to previous states where necessary.
+message Object {
+    // The object metadata that must be populated on both VersionVectors and Updates
+    string key = 1;                  // A unique key/id that represents the object across the namespace of the object type
+    string namespace = 2;            // The namespace of the object, used to disambiguate keys or different object types
+    Version version = 3;             // A version vector representing this objects current or latest version
+    string region = 4;               // The region code where the data originated
+    string owner = 5;                // The replica that created the object
+
+    // The object data that is only populated on Updates.
+    oneof data {
```

Comment:
Right now using `oneof`, which specifies the types that can be replicated and requires no additional parsing (plus is a bit more compact) - however, if we add data types that can be replicated, then we will have to change the protocol buffers. 

Alternative: use `Any` - which will allow us to put any object here and is more generic, but would contain extra information (the type URL) and would require parsing on the other end. Maybe not so bad though given that we have the namespace identifier? 

So what do we prefer, flexibility or type safety?